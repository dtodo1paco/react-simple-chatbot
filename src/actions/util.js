export function is_link(text) {
    if (!/^(f|ht)tps?:\/\//i.test(text)) {
        return false;
    }
    return true;
}
export function add_http(url) {
    if (!is_link()) {
        url = "http://" + url;
    }
    return url;
}