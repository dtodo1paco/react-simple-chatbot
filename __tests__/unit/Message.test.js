jest.autoMockOff();

import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';

import { generateMessage, generateMessageArray, generateWord } from '../util/fakeGenerator';
import { getRenderedContent, getComposite} from '../util/render.js';

const Message = require('components/Message').default;

describe('Messages testing...', () => {
    /** CUSTOM FOR COMPONENT **/
    const exampleSummary = generateMessage();
    const exampleErrors = generateMessageArray(generateWord().length);

    beforeEach(function() {
    });

    describe('render', () => {
        it('snapshot', () => {
            const wrapper = shallow(<Message
                success={false}
                summary="Snapshot summary"
                messages={["one message", "another message", "the third one"]} autohide={true} />
            );
            expect(wrapper).toMatchSnapshot();
        });
        it('when no success, message-error className is present', () => {
            let content = getRenderedContent(<Message
                success={false}
                summary={exampleSummary}
                messages={exampleErrors} autohide={true} />
            );
            expect(content.props.className).toBe('message-error');
        });
        it('when success, message-success className is present', () => {
            let content = getRenderedContent(<Message
                    success={true}
                    summary={exampleSummary}
                    messages={exampleErrors} autohide={true} />
            );
            expect(content.props.className).toBe('message-success');
        });

        it('message has the correct structure', () => {
            let content = getRenderedContent(<Message
                    success={false}
                    summary={exampleSummary}
                    messages={exampleErrors} autohide={true} />
            );
            expect(Array.isArray(content.props.children)).toBeTruthy();
            let summary = content.props.children[0];
            let messages = content.props.children[1];
            expect(typeof summary).toBe('string')
            expect(Array.isArray(messages)).toBeTruthy();
        });
        it('message has the correct summary [' + exampleSummary + "]", () => {
            let content = getRenderedContent(<Message
                    success={false}
                    summary={exampleSummary}
                    messages={exampleErrors} autohide={true} />
            );
            let summary = content.props.children[0];
            expect(summary).toBe(exampleSummary);
        });
        it('message has the correct ' + exampleErrors.length + " messages", () => {
            let content = getRenderedContent(<Message
                    success={false}
                    summary={exampleSummary}
                    messages={exampleErrors} autohide={true} />
            );
            let messages = content.props.children[1];
            expect(messages.length).toBe(exampleErrors.length);
            for (let i = 0; i < messages.length; i++) {
                expect(new String(messages[i])).toEqual(new String(<li>{exampleErrors[i]}</li>));
            }
        });
    });
});

/**
 console.log("Errors: " + renderedDOM().querySelectorAll('.visible').length);
**/