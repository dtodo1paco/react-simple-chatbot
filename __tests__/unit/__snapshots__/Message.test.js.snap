// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages testing... render snapshot 1`] = `
<div
  className="message autohide"
>
  <div
    className="message-error"
  >
    Snapshot summary
    <li
      key="0"
    >
      one message
    </li>
    <li
      key="1"
    >
      another message
    </li>
    <li
      key="2"
    >
      the third one
    </li>
  </div>
</div>
`;
